core/src/main/java/org/jruby/ext/ffi/StructLayout.java

```diff
@@ -658,7 +658,7 @@ public boolean equals(Object obj) {
 
         @Override
         public int hashCode() {
-            return 53 * 5 + (int) (this.offset ^ (this.offset >>> 32)) ^ type.hashCode();
+            return 53 * 5 + (int) (this.offset ^ (this.offset >>> 16)) ^ type.hashCode();
```

Comment:
This change fixes the `offset` int getting shifted to oblivion. However, I'd prefer to replace this with a hashCode generated by IntelliJ as follows:

```diff
diff --git a/core/src/main/java/org/jruby/ext/ffi/StructLayout.java b/core/src/main/java/org/jruby/ext/ffi/StructLayout.java
index f7e269d777..4d5bfdf0a8 100644
--- a/core/src/main/java/org/jruby/ext/ffi/StructLayout.java
+++ b/core/src/main/java/org/jruby/ext/ffi/StructLayout.java
@@ -658,7 +658,10 @@ public final class StructLayout extends Type {
 
         @Override
         public int hashCode() {
-            return 53 * 5 + (int) (this.offset ^ (this.offset >>> 32)) ^ type.hashCode();
+            int result = super.hashCode();
+            result = 31 * result + type.hashCode();
+            result = 31 * result + offset;
+            return result;
         }
 
         /**
```